### PR TITLE
append seed to config.json

### LINKED ISSUE #53: recover training: key error
When recovering prior vampire pertaining or classifier training, I encountered key errors:

Key [random_seed, pytorch_seed, numpy_seed] found in training configuration but not in the serialization directory we're recovering from.

----
Version: PyTorch 1.2.0, AllenNLP 0.8.4

Pretraining command:
```
export DATA_DIR="$(pwd)/examples/ag"
export VOCAB_SIZE=30000
python -m scripts.train \
            --config training_config/vampire.jsonnet \
            --serialization-dir model_logs/ag \
            --environment VAMPIRE \
            --device 0 \
            --seed 1 \
            --recover	 
```
Pretraining error:

```
2020-01-15 12:45:52,302 - INFO - allennlp.training.util - Recovering from prior training at model_logs/ag_1.
2020-01-15 12:45:52,324 - ERROR - allennlp.training.util - Key 'random_seed' found in training configuration but not in the serialization directory we're recovering from.
2020-01-15 12:45:52,324 - ERROR - allennlp.training.util - Key 'pytorch_seed' found in training configuration but not in the serialization directory we're recovering from.
2020-01-15 12:45:52,324 - ERROR - allennlp.training.util - Key 'numpy_seed' found in training configuration but not in the serialization directory we're recovering from.
```

Classification command:

```
export VAMPIRE_DIR="$(pwd)/model_logs/vampire"
export VAMPIRE_DIM=81
export THROTTLE=200
export EVALUATE_ON_TEST=1
python -m scripts.train \
            --config training_config/classifier.jsonnet \
            --serialization-dir model_logs/clf \
            --environment CLASSIFIER \
            --device 0 \
            --seed 1 \
	    --recover	
````

Classification error:

```
2020-01-15 12:52:46,094 - INFO - allennlp.training.util - Recovering from prior training at model_logs/clf_1.
2020-01-15 12:52:46,112 - ERROR - allennlp.training.util - Key 'pytorch_seed' found in training configuration but not in the serialization directory we're recovering from.
2020-01-15 12:52:46,112 - ERROR - allennlp.training.util - Key 'numpy_seed' found in training configuration but not in the serialization directory we're recovering from.
2020-01-15 12:52:46,112 - ERROR - allennlp.training.util - Key 'random_seed' found in training configuration but not in the serialization directory we're recovering from.
```

### PR DESCRIPTION
**Pull request by shuningjin**
- Fixes #53 
- Before recovering, append random_seed, pytorch_seed, numpy_seed to config.json

### CODE REVIEW

**Comment by kernelmachine** [comment]
Thanks for this! LGTM
